tests/python_package_test/test_engine.py

```diff
@@ -1507,40 +1507,235 @@ def test_get_split_value_histogram(self):
+        metrics_combination_train_lambdarank("lambdarank", best_iter_ndcg, True)
+        metrics_combination_train_lambdarank("l2", best_iter_l2, False)
+        metrics_combination_train_lambdarank("l2", best_iter_l2, True)
+        metrics_combination_train_lambdarank(["ndcg", "l2"], best_iter_min, False)
+        metrics_combination_train_lambdarank(["ndcg", "l2"], best_iter_ndcg, True)
+        metrics_combination_train_lambdarank(["l2", "ndcg"], best_iter_min, False)
+        metrics_combination_train_lambdarank(["l2", "ndcg"], best_iter_l2, True)
+
+        best_iter_ndcg = 11
+        best_iter_l2 = 1
+        best_iter_min = min([best_iter_ndcg, best_iter_l2])
+        metrics_combination_cv_lambdarank(None, best_iter_ndcg, False, True)
+        metrics_combination_cv_lambdarank(None, best_iter_ndcg, True, True)
+        metrics_combination_cv_lambdarank([], best_iter_ndcg, False, True)
+        metrics_combination_cv_lambdarank([], best_iter_ndcg, True, True)
+        metrics_combination_cv_lambdarank("ndcg", best_iter_ndcg, False, True)
+        metrics_combination_cv_lambdarank("ndcg", best_iter_ndcg, True, True)
+        metrics_combination_cv_lambdarank("lambdarank", best_iter_ndcg, False, True)
+        metrics_combination_cv_lambdarank("lambdarank", best_iter_ndcg, True, True)
+        metrics_combination_cv_lambdarank("l2", best_iter_l2, False, True)
+        metrics_combination_cv_lambdarank("l2", best_iter_l2, True, True)
+        metrics_combination_cv_lambdarank(["ndcg", "l2"], best_iter_min, False, True)
+        metrics_combination_cv_lambdarank(["ndcg", "l2"], best_iter_ndcg, True, True)
+        metrics_combination_cv_lambdarank(["l2", "ndcg"], best_iter_min, False, True)
+        metrics_combination_cv_lambdarank(["l2", "ndcg"], best_iter_l2, True, True)
+
+        metrics_combination_cv_lambdarank(None, best_iter_ndcg, False, False)
+        metrics_combination_cv_lambdarank(None, best_iter_ndcg, True, False)
+        metrics_combination_cv_lambdarank([], best_iter_ndcg, False, False)
+        metrics_combination_cv_lambdarank([], best_iter_ndcg, True, False)
+        metrics_combination_cv_lambdarank("ndcg", best_iter_ndcg, False, False)
+        metrics_combination_cv_lambdarank("ndcg", best_iter_ndcg, True, False)
+        metrics_combination_cv_lambdarank("lambdarank", best_iter_ndcg, False, False)
+        metrics_combination_cv_lambdarank("lambdarank", best_iter_ndcg, True, False)
+        metrics_combination_cv_lambdarank("l2", best_iter_l2, False, False)
+        metrics_combination_cv_lambdarank("l2", best_iter_l2, True, False)
+        metrics_combination_cv_lambdarank(["ndcg", "l2"], best_iter_min, False, False)
+        metrics_combination_cv_lambdarank(["ndcg", "l2"], best_iter_ndcg, True, False)
+        metrics_combination_cv_lambdarank(["l2", "ndcg"], best_iter_min, False, False)
+        metrics_combination_cv_lambdarank(["l2", "ndcg"], best_iter_l2, True, False)
```

Comment:
It's too much! All these cases are tested for regression application above. Here we want to test only `@` sign in metric name, and 1-2 tests should be enough for it.